eventsourcing/src/main/java/org/axonframework/eventsourcing/MultiStreamableMessageSource.java

```diff
@@ -0,0 +1,384 @@
+                                                       candidateMessagesToReturn.put(singleMessageSource.getKey(),
+                                                                                     new GenericTrackedSourcedEventMessage(
+                                                                                             singleMessageSource
+                                                                                                     .getKey(),
+                                                                                             trackedEventMessage)));
+            }
+        }
+
+        /**
+         * Calls close on each of the streams.
+         */
+        @Override
+        public void close() {
+            for (Map.Entry<String, BlockingStream<TrackedEventMessage<?>>> singleMessageSource : messageSources
+                    .entrySet()) {
+                singleMessageSource.getValue().close();
+            }
+        }
+    }
+
+
+    /**
+     * Builder class to instantiate a {@link MultiStreamableMessageSource}. The configurable filed
+     * {@code comparator}, which decides which message to process first if there is a choice defaults to the oldest
+     * message available (using the event's timestamp). The stream on which long polling is done for
+     * {@link MultiSourceMessageStream#hasNextAvailable(int, TimeUnit)} is also configurable.
+     */
+    public static class Builder {
+
+        private Comparator<TrackedSourcedEventMessage<?>> comparator = Comparator.comparing(EventMessage::getTimestamp);
+        private Map<String, StreamableMessageSource<TrackedEventMessage<?>>> messageSourceMap = new LinkedHashMap<>();
+        private String longPollingSource = "";
+
+        /**
+         * Adds a message source to the list of sources.
+         *
+         * @param messageSourceName a unique name identifying the stream.
+         * @param messageSource     the message source to be added.
+         * @return the current Builder instance, for fluent interfacing.
+         */
+        public Builder addMessageSource(String messageSourceName,
+                                        StreamableMessageSource<TrackedEventMessage<?>> messageSource) {
+            Assert.isFalse(messageSourceMap.containsKey(messageSourceName),
+                           () -> "the messageSource name must be unique");
+            Assert.isFalse(messageSourceMap.containsValue(messageSource), () -> "this message source is duplicated");
+
+            messageSourceMap.put(messageSourceName, messageSource);
+            return this;
+        }
+
+        /**
+         * Overrides the default comparator (which returns the oldest event available).
+         *
+         * @param comparator the comparator to use when deciding on which message to return.
+         * @return the current Builder instance, for fluent interfacing.
+         */
+        public Builder withComparator(Comparator<TrackedSourcedEventMessage<?>> comparator) {
```

Comment:
All the other `Builder` classes do not prepend the builder methods with `with...`. I'd thus suggest to rename this method to `comparator` in favor of consistency.